src/js/utils/environment.ts

```diff
@@ -35,3 +35,8 @@ export function getHermesVersion(): string | undefined {
     RN_GLOBAL_OBJ.HermesInternal.getRuntimeProperties()['OSS Release Version']
   );
 }
+
+/** Returns default environment based on __DEV__ */
+export function getDefaultEnvironment(): 'development' | 'production' {
+  return __DEV__ ? 'development' : 'production';
```

Comment:
Do we know 100% whether `__DEV__` is always defined? If not I would suggest doing:

```suggestion
  return typeof __DEV__ !== undefined && __DEV__ ? 'development' : 'production';
```

That way it doesn't throw if it isn't defined.

Reply:
doesn't typedef `__DEV__` returns 'undefined' instead of undefined? Additionally if `__DEV__` is not false, what's the expected return? fallback to the original empty string? if so maybe we could do the following
```suggestion
  return typeof __DEV__ !== 'undefined' ? (__DEV__ ? 'development' : 'production') :  '';
```

Reply:
I did a quick test and ` typeof __DEV__ !== undefined ` return true for undefined and false for 'undefined' so the code would keep breaking because it would pass to the second condition at any case.